### PR TITLE
Migrate to Model Registry V2

### LINKED ISSUE #120: Migrate to new Model Registry SDK

### PR DESCRIPTION
**Pull request by wduguay-air**
closes #120 

## Overview
This pull request migrates the current Model Registry implementation to the Model Registry V2 implementation introduced by the SDK in release [v2.5.0](https://gitlab.com/ai-r/cogment/cogment-py-sdk/-/tree/release/v2.5.0).

## Testing
The functional tests will be added to the PR after merging the Functional Test PR https://github.com/cogment/cogment-verse/pull/143.
- [x] Functional Test PR merged
- [x] Functional Tests added

## Other changes
- Modified some existing functional test configs to have a faster execution time.

### CODE REVIEW

File: cogment_verse/model_registry_v2.py [hunk 1]
```diff
@@ -0,0 +1,556 @@
+# Copyright 2022 AI Redefined Inc. <dev+cogment@ai-r.com>
```
Comment (Yves-air) [inline cogment_verse/model_registry_v2.py:1]: Why copying the cogment file? Why not use cogment?

Comment (Yves-air) [reply cogment_verse/model_registry_v2.py:1]: And why do you need the stub directly?

Comment (wduguay-air) [reply cogment_verse/model_registry_v2.py:1]: Alright, I'm totally for that.

**Comment by wduguay-air** [comment]
Since the last review/conversation, I removed the `model_registry` implemented locally in cogment-verse, in favor of entirely using the python sdk version. I also change the Model class to implement the same methods (`serialize_model`, `deserialize_model`) as other recent cogment projects.

**Comment by Yves-air** [comment]
Maybe you can have a look at some of the new functionality coming, to see if it would make sense to wait for the next version:
https://gitlab.com/ai-r/cogment/cogment-doc/-/merge_requests/155

**Comment by wduguay-air** [comment]
Anyways I was planning to wait a bit to wait for Ha to merge some agent implementations PRs that have been open for a long time. But for now, the current functionalities seem enough for the general cogment-verse use case. I will probably make a new PR in the near future to include them!

**Comment by Yves-air** [comment]
I would suggest you go through and change `model_version` for `model_iteration` everywhere

**Comment by wduguay-air** [comment]
I just made a pass to rename to `model_version` to `model_iteration`. For the code guidelines, I will create a separate task to make a pass over the entire repo in the near future.

File: actors/ppo_atari_pz.py [hunk 2]
```diff
@@ -188,10 +194,10 @@ async def impl(self, actor_session):
         action_space = environment_specs.get_action_space(seed=config.seed)
 
         # Get model
-        model, _, version_info = await actor_session.model_registry.retrieve_version(
-            PPOModel, config.model_id, config.model_version
-        )
-        log.info(f"Actor - retreved model number: {version_info['version_number']}")
+        serialized_model = await actor_session.model_registry.retrieve_model(config.model_id, config.model_version)
```
[the remark this answers is not in the document]
Comment (wduguay-air) [reply actors/ppo_atari_pz.py]: Done. I may rework on the way it is applied in the near future. Maybe move some of this kind a logic into the Model class.

File: actors/ppo_atari_pz.py [hunk 3]
```diff
@@ -1038,7 +1077,17 @@ def create_trial_params(trial_idx: int, iter_idx: int, actors: list):
 
                     # Publish the newly updated model
                     self.model.iter_idx = iter_idx
-                    version_info = await run_session.model_registry.publish_version(
-                        self.model, archived=num_updates % 50 == 0
-                    )
+                    serialized_model = PPOModel.serialize_model(self.model)
+
+                    archived = num_updates % 50 == 0
+                    if archived:
```
[the remark this answers is not in the document]
Comment (wduguay-air) [reply actors/ppo_atari_pz.py]: Done. For the hardcoded parameters. I will make a pass over all actors in a separate PR to move all of this to the configurations.

File: actors/ppo.py [hunk 4]
```diff
@@ -319,7 +319,12 @@ async def impl(self, actor_session):
         assert config.environment_specs.num_players == 1
 
         # Get model
-        serialized_model = await actor_session.model_registry.retrieve_model(config.model_id, config.model_iteration)
+        if config.model_iteration == -1:
+            serialized_model = await actor_session.model_registry.track_latest_model(config.model_id)
```
Comment (cloderic) [inline actors/ppo.py:323]: in that case you can directly provide `PPOModel.deserialize_model` to the function so that the deserialized model is cached.

Comment (wduguay-air) [reply actors/ppo.py]: Done

**Review by lhnguyen102 — APPROVED** [review]

**Pull request merged**